### DISCUSSION TITLE
Reset Trim Action

### DISCUSSION THREAD

**Discussion by inventor7777** [EdgeTX/edgetx#4929] [Ideas]
Hi all! 

I wanted to make a Special Function to reset a certain trim on my RadioMaster MT12 to its default value (0).

However, I don't believe there is a "Reset Trim __ to value __" action. 

It would be a nice feature to have in the future. Thanks!

**Comment by pfeerick** [comment]
Good pont. Some of the more "complex" stuff can be done through Lua, so you
could have a simple reset trim Lua that can be run by a SF ;)

**Reply by inventor7777** [reply]
How should I go about doing that? Is there a wiki about LUA scripts? I'd love to learn!

**Comment by pfeerick** [comment]
There are docs on the Lua API available in EdgeTX here :
https://luadoc.edgetx.org/

I would probably start looking at model.getFlightMode()
and model.setFlightMode() as trims are set as part of flight modes. You may
be lucky and find some scripts showing it in use if you search for that and
opentx.

**Reply by inventor7777** [reply]
Thank you!

**Reply by jrwieland** [reply]
There are existing ways in lua to set trims to a desired value in the lua API reference

Specifically

model.setFlightMode(index, params)

The full list of parameters that can be set are listed in the corresponding get.FlightMode description

https://luadoc.edgetx.org/part_iii_-_opentx_lua_api_reference/model-functions-less-than-greater-than-luadoc-begin-model/getflightmode

Lua is capable of doing many functions and performing complex tasks.   However, if the script is overly taxing on the radio resources it will be forced closed.
